flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionTest.java

```diff
@@ -82,202 +82,12 @@
-
-		ExecutionGraph executionGraph = ExecutionGraphTestUtils.createSimpleTestGraph(
-			slotProvider,
-			new NoRestartStrategy(),
-			jobVertex);
-
-		executionGraph.start(ComponentMainThreadExecutorServiceAdapter.forMainThread());
-
-		ExecutionJobVertex executionJobVertex = executionGraph.getJobVertex(jobVertexId);
-
-		final Execution execution = executionJobVertex.getTaskVertices()[0].getCurrentExecutionAttempt();
-
-		CompletableFuture<Execution> allocationFuture = execution.allocateResourcesForExecution(
-			executionGraph.getSlotProviderStrategy(),
-			LocationPreferenceConstraint.ALL,
-			Collections.emptySet());
-
-		assertTrue(allocationFuture.isDone());
-
-		assertEquals(ExecutionState.SCHEDULED, execution.getState());
-
-		assertEquals(slot, execution.getAssignedResource());
-
-		// cancelling the execution should move it into state CANCELED
-		execution.cancel();
-		assertEquals(ExecutionState.CANCELED, execution.getState());
-
-		assertEquals(slot, slotOwner.getReturnedSlotFuture().get());
-	}
-
-	/**
-	 * Tests that the slot is released in case of a execution cancellation when being in state
-	 * RUNNING.
-	 */
-	@Test
-	public void testSlotReleaseOnExecutionCancellationInRunning() throws Exception {
-		final JobVertex jobVertex = createNoOpJobVertex();
-		final JobVertexID jobVertexId = jobVertex.getID();
-
-		final SingleSlotTestingSlotOwner slotOwner = new SingleSlotTestingSlotOwner();
-
-		final LogicalSlot slot = createTestingLogicalSlot(slotOwner);
-
-		final ProgrammedSlotProvider slotProvider = new ProgrammedSlotProvider(1);
-		slotProvider.addSlot(jobVertexId, 0, CompletableFuture.completedFuture(slot));
-
-		ExecutionGraph executionGraph = ExecutionGraphTestUtils.createSimpleTestGraph(
-			slotProvider,
-			new NoRestartStrategy(),
-			jobVertex);
-
-		ExecutionJobVertex executionJobVertex = executionGraph.getJobVertex(jobVertexId);
-
-		final Execution execution = executionJobVertex.getTaskVertices()[0].getCurrentExecutionAttempt();
-
-		CompletableFuture<Execution> allocationFuture = execution.allocateResourcesForExecution(
-			executionGraph.getSlotProviderStrategy(),
-			LocationPreferenceConstraint.ALL,
-			Collections.emptySet());
-
-		assertTrue(allocationFuture.isDone());
-
-		assertEquals(ExecutionState.SCHEDULED, execution.getState());
-
-		assertEquals(slot, execution.getAssignedResource());
-
-		execution.deploy();
-
-		execution.switchToRunning();
-
-		// cancelling the execution should move it into state CANCELING
-		execution.cancel();
-		assertEquals(ExecutionState.CANCELING, execution.getState());
-
-		execution.completeCancelling();
-
-		assertEquals(slot, slotOwner.getReturnedSlotFuture().get());
-	}
-
-	/**
-	 * Tests that a slot allocation from a {@link SlotProvider} is cancelled if the
-	 * {@link Execution} is cancelled.
-	 */
-	@Test
-	public void testSlotAllocationCancellationWhenExecutionCancelled() throws Exception {
```

Comment:
I think this test is also still valid. At least I couldn't find a test which ensures that we cancel our slot allocations when we cancel  the execution. I think currently, this only happens if a task failed. But maybe the test need to make sure that the `DefaultScheduler` cancels all pending slot requests if the `SchedulerNG.cancel` is called.

Reply:
This test is outdated because it is testing the implementation of legacy method `allocateResourcesForExecution()`, while the DefaultScheduler has taken over the responsibility to allocate a slot for an execution. The cancellation of pending requests on vertex cancellation now locates in `DefaultScheduler#cancelExecutionVertex()`.
But I think you are right that now we lacks a test to verify that `ExecutionSlotAllocator#cancel()` is invoked when an execution vertex is canceled. I will add one in `DefaultSchedulerTest`.

Regarding `SchedulerNG.cancel()`, currently canceling a job does not directly cancel all pending requests in SlotPool. After the entire job is canceled and JobMaster is shutting down, all pending requests as well as allocated slots will be released when the SlotPool is closed. So I think the test needed is `SlotPoolImplTest#testShutdownCancelsAllPendingRequests()` which is missing at the moment. I will add it as well.

Reply:
added `DefaultSchedulerTest#allocationIsCanceledWhenVertexIsFailedOrCanceled` and `SlotPoolImpl#testShutdownCancelsAllPendingRequests`

Reply:
sounds good.